local/test-binary

```diff
@@ -92,46 +107,37 @@ if [[ "${DEFAULT_TESTING}" != "no" ]]; then
 
     process_mtr_output MySQL "${WORKDIR_ABS}/mtr.output" "${WORKDIR_ABS}/junit.xml"
 fi
-
-
+#
 if [[ "$HOTBACKUP_TESTING" != "no" ]] && [[ -n "${TOKUDB_PLUGIN}" ]] && [[ -n "${HOTBACKUP_LIB}" ]] && [[ -n "${HOTBACKUP_PLUGIN}" ]]; then
-    MYSQLD_ENV="${HOTBACKUP_LIB}:${JEMALLOC}:${EATMYDATA}"
-    if [[ "${ANALYZER_OPTS}" == *WITH_ASAN=ON* ]]; then
-        MYSQLD_ENV="$(find /opt /lib* /usr/lib* /usr/local/lib* -type f -name 'libasan.so*' | head -n1):${MYSQLD_ENV}"
-    fi
+    MYSQLD_ENV="${ADD_TO_LD_PRELOAD:-}${HOTBACKUP_LIB}:${JEMALLOC}:${EATMYDATA}"
 
-    LD_PRELOAD="${EATMYDATA} ${JEMALLOC}" MTR_BUILD_THREAD=auto ./mtr \
-        --force \
-        --max-test-fail=0 \
-        --suite-timeout=9999 \
-        --testcase-timeout=450 \
-        --parallel=$(grep -c ^processor /proc/cpuinfo) \
-        ${MTR_ARGS} \
-        --mysqld-env="LD_PRELOAD=${MYSQLD_ENV}" \
-        --suite tokudb.backup \
-        | tee ${WORKDIR_ABS}/tokudb_hotbackup.output \
-        || true
+    MTR_BUILD_THREAD=auto ./mtr \
+      --force \
+      --max-test-fail=0 \
+      --suite-timeout=9999 \
+      --testcase-timeout=450 \
+      --parallel=$(grep -c ^processor /proc/cpuinfo) \
+      ${MTR_ARGS} \
+      --mysqld-env="LD_PRELOAD=${MYSQLD_ENV}" \
+      --suite tokudb.backup \
+      | tee ${WORKDIR_ABS}/tokudb_hotbackup.output \
+      || true
     process_mtr_output "Percona Server - TokuDB HotBackup" "${WORKDIR_ABS}/tokudb_hotbackup.output" "${WORKDIR_ABS}/junit_tokudb_hotbackup.xml"
 fi
 
 # if there's tokudb plugin compiled run tokudb tests else exit with previous status
 if [[ "${TOKUDB_ENGINES_MTR}" = "yes" ]] && [[ -n "${TOKUDB_PLUGIN}" ]]; then
-    # we test with system jemalloc
-    if [[ -z "${JEMALLOC}" ]]; then
-        echo "No jemalloc lib found"
-        exit 1
-    fi
-
     # this is a workaround because mtr is looking for ha_tokudb.so in source_dir in lib/plugins/mysql when it's actually in builddir/storage/tokudb
-    mkdir -p lib/mysql/plugin
-    ln -s ${TOKUDB_PLUGIN}    lib/mysql/plugin/ha_tokudb.so
-    ln -s ${HOTBACKUP_PLUGIN} lib/mysql/plugin/tokudb_backup.so
+  [[ -d lib/mysql/plugin ]] || mkdir -p lib/mysql/plugin
```

Comment:
Why?

Reply:
if test is running on the same VM after build or test retry()'d- directory and links are already exist. it's just check - do not create directory if it exists, it's normal logic, nothing surprising

Reply:
What if it was a different version build on the same VM?